course/migrations/0110_add_force_lang_field_to_course.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.7 on 2017-12-20 10:09
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('course', '0109_add_manage_authentication_tokens_permssion'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='course',
+            name='force_lang',
+            field=models.CharField(blank=True, default=None, help_text="Which language name is forced to be used for this course. For example, 'en', 'de', 'zh-hans'. If not set, displayed language will be determined by user preference", max_length=200, null=True, verbose_name='Course language forcibly used'),
+        ),
+    ]
```

Comment:
I'm not totally opposed to this, but I'd like to understand the use case a bit better. What's the reason to not just trust the browser when it reports language preference to the server?

Reply:
For example, in some universities in China, there are some courses which are required to be taught in English ( including course materials), other than the native language.  It will be weird to have context (like date, menu and literals like "choices", "correct answers") in Chinese while the contents are in English.

Reply:
I see. Fair enough. :)